plugins/lookup/collection_version.py

```diff
@@ -0,0 +1,129 @@
+      - This can happen for collections installed from git which do not have a version number
+        in C(galaxy.yml).
+      - By default, C(*) is returned.
+    type: string
+    default: '*'
+"""
+
+EXAMPLES = """
+- name: Check version of community.general
+  ansible.builtin.debug:
+    msg: "community.general version {{ lookup('community.general.collection_version', 'community.general') }}"
+"""
+
+RETURN = """
+  _raw:
+    description:
+      - The version number of the collections listed as input.
+      - If a collection can not be found, it will return the value provided in I(result_not_found).
+        By default, this is C(none).
+      - If a collection can be found, but the version not identified, it will value provided in
+        I(result_no_version). By default, this is C(*). This can happen for collections installed
+        from git which do not have a version number in C(galaxy.yml).
+    type: list
+    elements: str
+"""
+
+import json
+import os
+import re
+
+import yaml
+
+from ansible.errors import AnsibleLookupError
+from ansible.module_utils.compat.importlib import import_module
+from ansible.plugins.lookup import LookupBase
+
+
+FQCN_RE = re.compile(r'^[A-Za-z0-9_]+\.[A-Za-z0-9_]+$')
+
+
+def load_collection_meta_manifest(manifest_path):
+    with open(manifest_path, 'rb') as f:
+        meta = json.load(f)
+    return {
+        'version': meta['collection_info']['version'],
+    }
+
+
+def load_collection_meta_galaxy(galaxy_path, no_version='*'):
+    with open(galaxy_path, 'rb') as f:
+        meta = yaml.safe_load(f)
+    return {
+        'version': meta.get('version') or no_version,
+    }
+
+
+class LookupModule(LookupBase):
+    def run(self, terms, variables=None, **kwargs):
+        result = []
+        self.set_options(var_options=variables, direct=kwargs)
+        not_found = self.get_option('result_not_found')
+        no_version = self.get_option('result_no_version')
+
+        for term in terms:
+            if not FQCN_RE.match(term):
+                raise AnsibleLookupError('"{term}" is not a FQCN'.format(term=term))
+
+            try:
+                collection_pkg = import_module('ansible_collections.{fqcn}'.format(fqcn=term))
+                if not collection_pkg:
+                    raise Exception('collection not found')
+            except Exception:
+                # Collection not found
+                result.append(not_found)
+                continue
+
+            try:
+                path = os.path.dirname(collection_pkg.__file__)
+                manifest_path = os.path.join(path, 'MANIFEST.json')
+                if os.path.exists(manifest_path):
+                    data = load_collection_meta_manifest(manifest_path)
+                else:
+                    data = {}
+                    galaxy_path = os.path.join(path, 'galaxy.yml')
+                    galaxy_alt_path = os.path.join(path, 'galaxy.yaml')
```

Comment:
As far as I can tell, `galaxy.yaml` was never a valid name for the collection metadata file (and `git grep` through ansible/ansible seems to confirm my suspicions). Maybe we can remove this line and simplify the code a bit?

Reply:
That's true for ansible-core 2.12 and Ansible 2.9, but ansible-base 2.10 and ansible-core 2.11 did support galaxy.yaml. Support was removed in https://github.com/ansible/ansible/commit/595413d11346b6f26bb3d9df2d8e05f2747508a3 (see the changes in `lib/ansible/galaxy/collection/__init__.py` which are hidden by default). I'll add a comment to document that though.

Reply:
Documented in 1169b5a01227615c3037cb6a5365ab6c7013a738.

I can also remove this, or make this code only execute for 2.10 and 2.11. What do you think?